runtime-light/tl/tl-types.h

```diff
@@ -304,4 +312,148 @@ struct confdataValue final {
   bool fetch(TLBuffer &tlb) noexcept;
 };
 
+// ===== HTTP =====
+
+struct HttpVersion final {
+  enum class Version : uint32_t {
+    Invalid = TL_ZERO,
+    V09 = 0x9e15'd325,
+    V10 = 0xfed2'5ac0,
+    V11 = 0x89d5'6a56,
+    V2 = 0x9022'9734,
+    V3 = 0xe725'a7a2,
+  };
+
+  Version version{Version::Invalid};
+
+  bool fetch(TLBuffer &tlb) noexcept {
+    using version_utype = std::underlying_type_t<Version>;
+
+    switch (tlb.fetch_trivial<uint32_t>().value_or(TL_ZERO)) {
+      case static_cast<version_utype>(Version::V09): {
```

Comment:
I would prefer to get rid of `enum class`, just `enum`. And here wouldn't be awful `static_casts`-s.
But up to you

Reply:
I always prefer enum classes due to their safety